common/server_protocolhandler.cpp

```diff
@@ -542,6 +542,24 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
     if (!r)
         return Response::RespNameNotFound;
 
+    QString roomPermission = r->getRoomPermission().toLower();
+    if (roomPermission != "none"){
+        if (roomPermission == "registered") {
```

Comment:
Also, you should probably save `userInfo->user_level()` as a variable since you call it multiple times in this block